Multiple crash found by fuzzing

PR: https://github.com/osrg/gobgp/pull/2707

### 1

file:  `gobgp/pkg/packet/rtr/rtr_test.go`

```go
func TestParseRTRbug(t *testing.T) {
	ParseRTR([]byte("0\x0400000000"))
}
```

run: `go test -run TestParseRTRbug 	$(pwd)/pkg/packet/rtr`


backtrace:
```log
--- FAIL: TestParseRTRbug (0.00s)
panic: runtime error: index out of range [10] with length 10 [recovered]
	panic: runtime error: index out of range [10] with length 10

goroutine 6 [running]:
testing.tRunner.func1.2({0x6439e0, 0xc00001e390})
	/usr/lib/golang/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
	/usr/lib/golang/src/testing/testing.go:1529 +0x39f
panic({0x6439e0, 0xc00001e390})
	/usr/lib/golang/src/runtime/panic.go:884 +0x213
github.com/osrg/gobgp/v3/pkg/packet/rtr.(*RTRIPPrefix).DecodeFromBytes(0x10?, {0xc00001a960?, 0xc00001a960?, 0x0?})
	/home/legion/0x34d/gobgp/pkg/packet/rtr/rtr.go:217 +0x18c
github.com/osrg/gobgp/v3/pkg/packet/rtr.ParseRTR({0xc00001a960, 0xa, 0xa})
	/home/legion/0x34d/gobgp/pkg/packet/rtr/rtr.go:390 +0x138
github.com/osrg/gobgp/v3/pkg/packet/rtr.TestParseRTRbug(0x0?)
	/home/legion/0x34d/gobgp/pkg/packet/rtr/rtr_test.go:127 +0x45
testing.tRunner(0xc0000d0820, 0x686448)
	/usr/lib/golang/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	/usr/lib/golang/src/testing/testing.go:1629 +0x3ea
FAIL	github.com/osrg/gobgp/v3/pkg/packet/rtr	0.008s
FAIL
```

--- a/pkg/packet/bgp/bgp.go
+++ b/pkg/packet/bgp/bgp.go
@@ -1013,10 +1013,26 @@ func (c *CapFQDN) DecodeFromBytes(data []byte) error {
 	if len(data) < 2 {
 		return NewMessageError(BGP_ERROR_OPEN_MESSAGE_ERROR, BGP_ERROR_SUB_UNSUPPORTED_CAPABILITY, nil, "Not all CapabilityFQDN bytes allowed")
 	}
+	rest := len(data)
+	if rest < 1 {
+		return NewMessageError(BGP_ERROR_OPEN_MESSAGE_ERROR, BGP_ERROR_SUB_UNSUPPORTED_CAPABILITY, nil, "Not all CapabilityFQDN bytes allowed")
+	}
 	hostNameLen := uint8(data[0])
+	rest -= 1
 	c.HostNameLen = hostNameLen
+	if rest < int(hostNameLen) {
+		return NewMessageError(BGP_ERROR_OPEN_MESSAGE_ERROR, BGP_ERROR_SUB_UNSUPPORTED_CAPABILITY, nil, "Not all CapabilityFQDN bytes allowed")
+	}
 	c.HostName = string(data[1 : c.HostNameLen+1])
+	rest -= int(hostNameLen)
+	if rest < 1 {
+		return NewMessageError(BGP_ERROR_OPEN_MESSAGE_ERROR, BGP_ERROR_SUB_UNSUPPORTED_CAPABILITY, nil, "Not all CapabilityFQDN bytes allowed")
+	}
+	rest -= 1
 	domainNameLen := uint8(data[c.HostNameLen+1])
+	if rest < int(domainNameLen) {
+		return NewMessageError(BGP_ERROR_OPEN_MESSAGE_ERROR, BGP_ERROR_SUB_UNSUPPORTED_CAPABILITY, nil, "Not all CapabilityFQDN bytes allowed")
+	}
 	c.DomainNameLen = domainNameLen
 	c.DomainName = string(data[c.HostNameLen+2:])
 	return nil

